File: tasks/expr-eval/python-typed/environment/src/evaluate.py
"""Evaluation of the tokens the current parser produces."""

from typing import List, cast

from parse import Token


def evaluate(tokens: List[Token]) -> int:
    """Left to right, with no precedence between the operators."""
    value = cast(int, tokens[0][0])
    index = 1
    while index < len(tokens):
        operator = cast(str, tokens[index][0])
        right = cast(int, tokens[index + 1][0])
        value = value + right if operator == "+" else value * right
        index += 2
    return value

File: tasks/expr-eval/javascript/environment/src/evaluate.js
"use strict";

/** Left to right, with no precedence between the operators. */
function evaluate(tokens) {
  let value = tokens[0].value;
  let index = 1;
  while (index < tokens.length) {
    const operator = tokens[index].value;
    const right = tokens[index + 1].value;
    value = operator === "+" ? value + right : value * right;
    index += 2;
  }
  return value;
}

module.exports = { evaluate };
